manager/apis/app/v1alpha1/plotter_types.go

```diff
@@ -4,21 +4,223 @@
 package v1alpha1
 
 import (
+	"fybrik.io/fybrik/pkg/serde"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// AssetDetails is a list of assets used in the fybrikapplication. In addition to assets declared in
+// fybrikapplication, AssetDetails list also contains assets that are allocated by the control-plane
+// in order to serve fybrikapplication
+type AssetDetails struct {
+	// AdvertisedAssetID links this asset to asset from fybrikapplication and is used by user facing services
+	// +optional
+	AdvertisedAssetID string `json:"advertisedAssetId,omitempty"`
+
+	// +required
+	DataStore DataStore `json:"assetDetails"`
+}
+
+type Service struct {
+	//+required
+	Endpoint EndpointSpec `json:"endpoint"`
+
+	// Format represents data format (e.g. parquet) as received from catalog connectors
+	// +required
+	Format string `json:"format"`
+}
+
+// StepSource is the source of this step: it could be assetID
+// or an enpoint of another step
+type StepSource struct {
+	// AssetID identifies the source asset of this step
+	// +optional
+	AssetID string `json:"assetId,omitempty"`
+
+	//+optional
+	API *Service `json:"api,omitempty"`
+}
+
+// StepSink holds information to where the target data will be written:
+// it could be assetID of an asset specified in fybrikapplication or of an asset created
+// by fybrik control-plane
+type StepSink struct {
+	// AssetID identifies the target asset of this step
+	// +required
+	AssetID string `json:"assetId"`
+}
+
+// StepParameters holds the parameters to the module
+// that is deployed in this step
+type StepParameters struct {
+	// +optional
+	Source *StepSource `json:"source,omitempty"`
+
+	// +optional
+	Sink *StepSink `json:"sink,omitempty"`
+
+	// +optional
+	API *Service `json:"api,omitempty"`
+
+	// Actions are the data transformations that the module supports
+	// +optional
+	Actions []serde.Arbitrary `json:"action,omitempty"`
+}
+
+// DataFlowStep contains details on a single data flow step
+type DataFlowStep struct {
+	// Name of the step
+	// +required
+	Name string `json:"name"`
+
+	// Name of the cluster this step is executed on
+	// +required
+	Cluster string `json:"cluster"`
+
+	// Template is the name of the template to execute the step
+	// The full details of the template can be extracted from Plotter.spec.templates
+	// list field.
+	// +required
+	Template string `json:"template"`
+
+	// Step parameters
+	// TODO why not flatten the parameters into this data flow step
+	// +optional
+	Parameters *StepParameters `json:"parameters,omitempty"`
+}
+
+// SubFlowTrigger indicates the trigger for this subflow
+// +kubebuilder:validation:Enum=workload;init;timer
+type SubFlowTrigger string
+
+const (
+	// Copy flow trigger
+	CopyTrigger SubFlowTrigger = "init"
```

Comment:
I see you changed the values.  Please change the names of the variables and update the comments as well.  TODO comment: These will come from the taxonomy in the future.

Reply:
Fixed